src/UnrollLoops.cpp

```diff
@@ -80,6 +82,75 @@ class UnrollLoops : public IRMutator {
                 user_warning << "Warning: Unrolling a for loop of extent 1: " << for_loop->name << "\n";
             }
 
+            // Peel lets that don't depend on the loop var to avoid needlessly
```

Comment:
Could we just do LICM first, and then not need this code?

Reply:
That was my first attempt. If you unroll an outer loop though it LICMs all inner loops, which is a much larger change than I was comfortable with. LICM also doesn't lift loads, which I want to. LICM needs an overhaul in general to do a proper graph cut instead of the current heuristic. It makes bad decisions about what to lift.